ddtrace/contrib/kafka/patch.py

```diff
@@ -149,7 +162,15 @@ def traced_produce(func, instance, args, kwargs):
         span.set_tag_str(COMPONENT, config.kafka.integration_name)
         span.set_tag_str(SPAN_KIND, SpanKind.PRODUCER)
         span.set_tag_str(kafkax.TOPIC, topic)
-        span.set_tag_str(kafkax.MESSAGE_KEY, ensure_text(message_key, errors="replace"))
+
+        if isinstance(instance, _SerializingProducer):
```

Comment:
`isinstance(instance, None)` will raise a TypeError

```suggestion
        if _SerializingProducer is not None and isinstance(instance, _SerializingProducer):
```